### PR TITLE
Docs RTIO sequencing a bit

### PR DESCRIPTION
**Pull request by charlesbaynham**
#ARTIQ Pull Request

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :scroll: Docs |

### All Pull Requests

- [x] Use correct spelling and grammar.
- [x] Check the copyright situation of your changes and sign off your patches (`git commit --signoff`, see [copyright](../CONTRIBUTING.rst#copyright-and-sign-off)).

### Documentation Changes

Add a couple of sentences explaining how SED sequencing works and how sequencing errors can occur. 

### Licensing

See [copyright & licensing for more info](https://github.com/m-labs/artiq/blob/master/CONTRIBUTING.rst#copyright-and-sign-off).
ARTIQ files that do not contain a license header are copyrighted by M-Labs Limited and are licensed under LGPLv3+.

### CODE REVIEW

File: doc/manual/rtio.rst [hunk 1]
```diff
@@ -123,14 +123,15 @@ Sequence errors
...
 
 Notes:
 
 * Strictly increasing timestamps never cause sequence errors. 
 * Configuring the gateware with more lanes for the RTIO core reduces the frequency of sequence errors.
 * The number of lanes is a hard limit on the number of simultaneous RTIO output events.
 * Whether a particular sequence of timestamps causes a sequence error or not is fully deterministic (starting from a known RTIO state, e.g. after a reset). Adding a constant offset to the whole sequence does not affect the result.
+* Zero-duration methods (such as :meth:`artiq.coredevice.ttl.TTLOut.on()`) do not advance the timeline and so will consume additional lanes if they are scheduled simultaneously. Adding a tiny delay will prevent this (e.g. ``delay_mu(1)``).
```
Comment (jordens) [inline doc/manual/rtio.rst:134]: This needs to be at least one coarse rtio cycle, so `delay_mu(8)` typically.

Comment (charlesbaynham) [reply doc/manual/rtio.rst:134]: Huh in my dumb experiment adding just 1 mu seemed to do the trick:

```
class TestSeq(EnvExperiment):

    def build(self):
        self.setattr_device("core")
        self.setattr_device("ttl4")
        self.setattr_device("ttl5")
        self.setattr_device("ttl6")
        self.setattr_device("ttl7")
        self.setattr_device("ttl8")
        self.setattr_device("ttl9")
        self.setattr_device("ttl10")
        self.setattr_device("ttl11")
        self.setattr_device("ttl12")
        self.setattr_device("ttl13")

        self.ttls = [
            self.ttl4,
            self.ttl5,
            self.ttl6,
            self.ttl7,
            self.ttl8,
            self.ttl9,
            self.ttl10,
            self.ttl11,
            self.ttl12,
            self.ttl13,
        ]

    @kernel
    def run(self):

        self.core.reset()

        for _ in range(10):
            for ttl in self.ttls:
                ttl.on()
                delay_mu(1)
```

Comment (charlesbaynham) [reply doc/manual/rtio.rst:134]: I'll change the PR. Do you know why this worked?

Comment (sbourdeauducq) [reply doc/manual/rtio.rst]: Likely because ``now_mu mod 8`` happened to be 7 when you ran that.

Comment (sbourdeauducq) [reply doc/manual/rtio.rst]: > This needs to be at least one coarse rtio cycle, so `delay_mu(8)` typically.

``delay(self.core.ref_multiplier)``

File: doc/manual/rtio.rst [hunk 2]
```diff
@@ -123,14 +123,15 @@ Sequence errors
 ---------------
 A sequence error happens when the sequence of coarse timestamps cannot be supported by the gateware. For example, there may have been too many timeline rewinds.
 
-Internally, the gateware stores output events in an array of FIFO buffers (the "lanes") and the timestamps in each lane much be strictly increasing. The gateware selects a different lane when an event with a decreasing or equal timestamp is submitted. A sequence error occurs when no appropriate lane can be found.
+Internally, the gateware stores output events in an array of FIFO buffers (the "lanes") and the timestamps in each lane must be strictly increasing. If an event with a decreasing or equal timestamp is submitted, the gateware selects the next lane, wrapping around if the final lane is reached. If this lane also contains an event with a timestamp beyond the one being submitted then a sequence error occurs. See `this issue <https://github.com/m-labs/artiq/issues/1081>`_ for a real-life example of how this works.
```
Comment (sbourdeauducq) [inline doc/manual/rtio.rst:126]: It would be better if the docs were self-contained and did not reference GitHub.

**Pull request merged**